apps/nextjs-app-router/src/app/[lang]/[[...path]]/page.tsx

```diff
@@ -2,12 +2,11 @@ import { client } from '@/makeswift/client'
 import '@/makeswift/components'
 import { getSiteVersion } from '@makeswift/runtime/next/server'
 import { notFound } from 'next/navigation'
-import { Page as MakeswiftPage } from '@makeswift/runtime/next'
+import { Page as MakeswiftPage } from '@makeswift/runtime/next/app-router'
```

Comment:
Breaking `runtime` API change: had to introduce a separate, App Router-specific entry point for client-side `runtime` imports, see below.

Reply:
Instead of naming it `app-router`, what if we named the import `server` or something that's not specifically `app-router`?? This seems to be a normal distinction that's done in Next.js itself and other popular Next.js libraries.

Examples:
- https://clerk.com/docs/references/nextjs/clerk-middleware
- https://nextjs.org/docs/app/building-your-application/routing/middleware#example
- https://next-intl-docs.vercel.app/docs/getting-started/app-router/with-i18n-routing#i18n-request

Reply:
Also, what was the particular reason we had to introduce a separate entry point? For imports that are client-side, shouldn't we be able to have `use client` directives that avoid that problem?

Reply:
@migueloller So the reason we have to introduce an App Router-specific endpoint is that importing and extending `Document` from `@next/document` [breaks App Router's production build](https://vercel.com/makeswift/makeswift-nextjs-app-router/AFZqEjwe3A1yZMQd1YaMuw1A9ZGG?filter=errors), and our default "client" Next.js entry point, `@makeswift/runtime/next`, has to export the (extended) `Document` for the Pages Router.

We could extend [the already existing `@makeswift/runtime/next/server` endpoint](https://github.com/makeswift/makeswift/blob/main/packages/runtime/package.json#L61) with the client-only exports we need for the App Router (`RootStyleRegistry` and `ReactRuntimeProvider`), but then it would be misnamed.

Reply:
Converged on splitting out the `Document` export into its own entry point, `@makeswift/runtime/next/document`, so we can keep the `@makeswift/runtime/next` as our primary client-side import for the App Router.